src/datasets.c

```diff
@@ -1224,6 +1539,30 @@ static DataRepResultType DatasetLookupIPv6wRep(
     return rrep;
 }
 
+static DataJsonResultType DatasetLookupIPv6wJson(
+        Dataset *set, const uint8_t *data, const uint32_t data_len)
+{
+    DataJsonResultType rrep = { .found = false, .json = { .value = NULL, .len = 0 } };
+
+    if (set == NULL)
+        return rrep;
+
+    if (data_len != 16 && data_len != 4)
```

Comment:
how would we get here with data_len == 4?

Reply:
GetDataDst and GetDataSrc  in detect-ipaddr.c are setting a buffer length of 4 or 16 depending of IP type:
```C
static InspectionBuffer *GetDataDst(DetectEngineThreadCtx *det_ctx,
        const DetectEngineTransforms *transforms, Packet *p, const int list_id)
{
    InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
    if (buffer->inspect == NULL) {
        if (PacketIsIPv4(p)) {
            /* Suricata stores the IPv4 at the beginning of the field */
            InspectionBufferSetup(det_ctx, list_id, buffer, p->dst.address.address_un_data8, 4);
        } else if (PacketIsIPv6(p)) {
            InspectionBufferSetup(det_ctx, list_id, buffer, p->dst.address.address_un_data8, 16);
```
so we can have the 2 lengths. As Suricata stores the IPv4 at the beginning of the field, this is working.